src/utils/Deprecation.ts

```diff
@@ -0,0 +1,18 @@
+export class Deprecation {
+
+  private _deprecationWarning: boolean;
+
+  constructor (deprecationWarning: boolean) {
+    this._deprecationWarning = deprecationWarning;
+  }
+
+  logDeprecation (response) {
+    if (this._deprecationWarning && response.deprecations?.length) {
+      response.deprecations.forEach(deprecation => {
```

Comment:
Usually we prefer the `for (const .. of )` loop:
 - `forEach` take a function so if you return inside the loop then you don't exit the main function
 - if you await a promise inside the function or return a promise, it's not awaited before starting calling the function on the next item
 
 Because of those two reason it's better to always have `for (const .. of ` loop for the sack of consistency

Reply:
Good to know!